nmdc_runtime/site/changesheets/scripts/missing_neon_soils_ecosystem_data.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python
+# nmdc_runtime/site/changesheets/scripts/missing_neon_soils_ecosystem_data.py
+"""
+missing_neon_soils_ecosystem_data.py: Create a changesheet for missing ecosystem data for NEON soils samples
+"""
+
+import logging
+import os
+import time
+from pathlib import Path
+from typing import List, Tuple
+
+import click
+from dotenv import load_dotenv
+
+from nmdc_runtime.site.changesheets.base import (
+    Changesheet,
+    ChangesheetLineItem,
+    JSON_OBJECT,
+    CHANGESHEETS_DIR,
+)
+from nmdc_runtime.site.normalization.gold import (
+    get_gold_biosample_name_suffix,
+    normalize_gold_id,
+)
+from nmdc_runtime.site.resources import GoldApiClient, RuntimeApiUserClient
+
+load_dotenv()
+GOLD_NEON_SOIL_STUDY_ID = "Gs0144570"
+NAME = "missing_neon_soils_ecosystem_data"
+# omics processing to biosamples file in data directory
+DATA_PATH = Path(__file__).parent.parent.joinpath("data")
+
+log_filename = f"{NAME}-{time.strftime('%Y%m%d-%H%M%S')}.log"
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s %(message)s",
+    filename=log_filename,
+    encoding="utf-8",
+    filemode="w",
+)
+
+
+def read_omics_processing_to_biosample_map() -> dict:
+    """
+    Map omics processing to biosamples
+    :return: dict
+    """
+    omics_processing_to_biosamples = {}
+    with open(DATA_PATH.joinpath("OmicsProcessing-to-catted-Biosamples.tsv")) as f:
+        lines_read = 0
+        lines_skipped = 0
+        for line in f:
+            if not line.startswith("nmdc:"):
+                lines_skipped += 1
+                continue
+            lines_read += 1
+            omics_processing_id, biosample_ids = line.strip().split("\t")
+            omics_processing_to_biosamples[omics_processing_id] = biosample_ids.split(
+                "|"
+            )
+    logging.debug(
+        f"Read {lines_read} lines from OmicsProcessing-to-catted-Biosamples.tsv"
+    )
+    logging.info(
+        f"Skipped {lines_skipped} lines from OmicsProcessing-to-catted-Biosamples.tsv"
+    )
+    return omics_processing_to_biosamples
+
+
+def goldbs_to_nmdcbs_and_omprc(
```

Comment:
Since this function's name contains some abbreviations that I'm not familiar with (took me a few seconds to translate them into "gold biosamples to NMDC biosamples and omics processings"), I'd prefer the function name be more "spelled out" or at least have a docstring that explains its functionality.

Reply:
why not both?